### PR TITLE
feat: config page ux improvements

### LINKED ISSUE #290: ux: make URL lists line-separated textboxes
## Problem

JSON array in input is bad UX, hard to edit, and very error prone:

## Proposed fix

Do what we do in ipfs-companion, have newline-separated textarea that grows a bit based on the content:

> 

>

### PR DESCRIPTION
**Pull request by SgtPooki**
- feat: convert config inputs to textarea
- fix: save config works
- feat: add descriptions to config page inputs
- chore: package-lock.json update

## Title

feat: config page ux improvements

## Description

Updates the config page inputs that were `type=input` to `<textarea>` and adds descriptions to each input. It makes the page more user-friendly and less prone to input errors.

Some style updates were done to resolve a long-standing UX issue with error message placement.

Fixes https://github.com/ipfs/service-worker-gateway/issues/290

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/lib/config-db.ts [hunk 1]
```diff
@@ -23,14 +23,14 @@ export async function loadConfigFromLocalStorage (): Promise<void> {
   if (typeof globalThis.localStorage !== 'undefined') {
     await configDb.open()
     const localStorage = globalThis.localStorage
-    const localStorageGatewaysString = localStorage.getItem(LOCAL_STORAGE_KEYS.config.gateways) ?? JSON.stringify(defaultGateways)
-    const localStorageRoutersString = localStorage.getItem(LOCAL_STORAGE_KEYS.config.routers) ?? JSON.stringify(defaultRouters)
-    const localStorageDnsResolvers = localStorage.getItem(LOCAL_STORAGE_KEYS.config.dnsJsonResolvers) ?? JSON.stringify(defaultDnsJsonResolvers)
+    const localStorageGatewaysString = localStorage.getItem(LOCAL_STORAGE_KEYS.config.gateways) ?? convertUrlArrayToInput(defaultGateways)
```
Comment (lidel) [inline src/lib/config-db.ts]: nit: would it be possible to keep existing behavior (store JSON) and convert to prettier newline delimited representation only in the end users GUI?

Right now, existing users that have preexisting JSON in storage are being broken (screenshot below):

>

**Review by lidel — CHANGES_REQUESTED** [review]
Thanks! Pushed small css tweaks, validation now looks like this:

> 

Small concern about changing storage format (see details below), other than that lgtm.

File: src/lib/config-db.ts (hunk 1 above)
Comment (SgtPooki) [reply src/lib/config-db.ts]: totally. let me fiddle with that. i created functions for conversion both ways so that shouldn't be a big deal.

Comment (SgtPooki) [reply src/lib/config-db.ts]: ok @lidel give it a try with [add703b](https://github.com/ipfs/service-worker-gateway/pull/315/commits/add703b9e46fc13436ef79498835e1d37323cb33)

**Review by lidel — APPROVED** [review]
Thank you @SgtPooki, let's ship this :+1:

**Pull request merged**